src/app/qgsdecorationcopyrightdialog.cpp

```diff
@@ -82,6 +87,30 @@ void QgsDecorationCopyrightDialog::buttonBox_rejected()
   reject();
 }
 
+void QgsDecorationCopyrightDialog::mInsertExpressionButton_clicked()
+{
+  QString selText = txtCopyrightText->textCursor().selectedText();
+
+  // edit the selected expression if there's one
+  if ( selText.startsWith( QLatin1String( "[%" ) ) && selText.endsWith( QLatin1String( "%]" ) ) )
+    selText = selText.mid( 2, selText.size() - 4 );
+
+  QgsExpressionContext expressionContext = QgsExpressionContext();
+  expressionContext.appendScope( QgsExpressionContextUtils::globalScope() );
+  expressionContext.appendScope( QgsExpressionContextUtils::projectScope( QgsProject::instance() ) );
```

Comment:
again, i'd suggest taking the expression context from the canvas's mapsettings instead